helix-tui/src/text.rs

```diff
@@ -229,6 +229,70 @@ impl<'a> Spans<'a> {
     pub fn width(&self) -> usize {
         self.0.iter().map(Span::width).sum()
     }
+
+    /// Constructs a styled `Span` from the given `string` and `style`
+    /// and appends it the to the `Spans`.
+    ///
+    /// ## Examples
+    ///
+    /// ```rust
+    /// # use helix_tui::text::{Span, Spans};
+    /// # use helix_view::graphics::{Color, Style};
+    /// let mut spans = Spans::new();
+    /// spans.push("Hello!", Style::default());
+    /// assert_eq!("Hello!", spans.0[0].content);
+    /// ```
+    pub fn push<T>(&mut self, string: T, style: Style)
+    where
+        T: Into<Cow<'a, str>>,
+    {
+        self.0.push(Span::styled(string, style));
+    }
+
+    /// Creates a new, empty `Spans`.
+    pub fn new() -> Spans<'a> {
+        Spans(vec![])
+    }
+
+    /// Constructs a single, styled `Span` with the given `string` and `style`
+    /// and wraps it in a `Spans`.
+    pub fn single<T>(string: T, style: Style) -> Spans<'a>
+    where
+        T: Into<Cow<'a, str>>,
+    {
+        Spans(vec![Span::styled(string, style)])
+    }
```

Comment:
This already exists as a `From`: https://github.com/helix-editor/helix/blob/2a11fb485d33eb33a7743899e9a86f745f6d5e02/helix-tui/src/text.rs#L258-L262

Places where this function is used can either use `Span::styled(string, style).into()` or explicitly call `Spans::from` on that Span.

Reply:
Didn't know, will change. Thanks!

Reply:
Fixed in 560fd9aa